src/jdk.httpserver/share/classes/com/sun/net/httpserver/Headers.java

```diff
@@ -254,4 +284,57 @@ public String toString() {
         sb.append(" }");
         return sb.toString();
     }
+
+    private static final Headers EMPTY = new UnmodifiableHeaders(new Headers());
```

Comment:
IDEA warns here:
>Referencing subclass UnmodifiableHeaders from superclass Headers initializer might lead to class loading deadlock
>Such references can cause JVM-level deadlocks in multithreaded environment, when one thread tries to load the superclass and another thread tries to load the subclass at the same time.

Reply:
Interesting, thanks for noting. We can return a new instance instead on line 307, the only place the constant is used.

Reply:
Or you could possibly define the constant in UnmodifiableHeaders instead.